### PR TITLE
[WNMGDS-2529] Give the calendar button more context for screen readers

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

WNMGDS-2529

Sometimes there are multiple date pickers on the same page. Screen reader users need to be able to distinguish between the different calendar buttons.

- Give the calendar button more context for screen readers by describing by the label and hint

## How to test

[Branch demo](https://cmsgov.github.io/design-system/branch/pwolfert/give-calendar-button-sr-context/storybook)

1. Open the [with picker](https://cmsgov.github.io/design-system/branch/pwolfert/give-calendar-button-sr-context/storybook/?path=/story/components-singleinputdatefield--with-picker) story in Storybook
2. Using a screen reader, focus on the calendar button. It should read out the label and hint after a short pause to give some extra context

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to code:

- [x] Created or updated unit tests to cover any new or modified code
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### CODE REVIEW

File: packages/design-system/src/components/DateField/SingleInputDateField.tsx [hunk 1]
```diff
@@ -166,6 +166,9 @@ const SingleInputDateField = (props: SingleInputDateFieldProps) => {
             onClick={() => setPickerVisible(!pickerVisible)}
             type="button"
             ref={calendarButtonRef}
+            // The `?? ''` after `hintId` is only to support v8.0, which doesn't have a `hintId`.
+            // It can be removed after we're done supporting v8.0.
+            aria-describedby={`${labelProps.id} ${labelProps.hintId ?? ''}`}
```
[the remark this answers is not in the document]
Comment (pwolfert) [reply packages/design-system/src/components/DateField/SingleInputDateField.tsx:171]: This is what it says:

> Open calendar picker dialog, button
>
> What day did you move? This date should be within the past 60 days in order to qualify, You are currently on a button, inside of web content. To click this button, press

Are you suggesting that "Open calendar picker dialog" be changed to something else, @zarahzachz?

[the remark this answers is not in the document]
Comment (pwolfert) [reply packages/design-system/src/components/DateField/SingleInputDateField.tsx:171]: Dan and I talked about what to add to this before I made the change, which was to add the aria-describedby of the label/hint. We could also talk about refining the text of the button though

[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/design-system/src/components/DateField/SingleInputDateField.tsx:171]: oh i didnt hear the "open calendar picker dialog" part. disregard then. i just heard the label + hint

[the remark this answers is not in the document]
Comment (zarahzachz) [reply packages/design-system/src/components/DateField/SingleInputDateField.tsx:171]: retested and i must've missed the dialog part. it works.

**Pull request merged**